File: ws/src/main/scala/de/thm/ii/fbs/model/CourseRole.scala
package de.thm.ii.fbs.model

/**
  * The course roles of a user
  */
object CourseRole extends Enumeration {
  /**
    * Admin of the course.
    */
  val DOCENT: CourseRole.Value = Value(0)
  /**
    * Moderator of the course
    */
  val TUTOR: CourseRole.Value = Value(1)
  /**
    * Participant in the course.
    */
  val STUDENT: CourseRole.Value = Value(2)

  /**
    * Parse an int to a course role.
    * @param roleId The role id
    * @return Course role, where the id is mapped to its assigned enum type, if no enum type
    *         is assigned to the roleId provided, then the STUDENT type is returned.
    */
  def parse(roleId: Int): CourseRole.Value = roleId match {
    case 0 => CourseRole.DOCENT
    case 1 => CourseRole.TUTOR
    case _ => CourseRole.STUDENT
  }
}

File: modules/fbs-core/api/src/main/kotlin/de/thm/ii/fbs/model/v2/CourseRole.kt
package de.thm.ii.fbs.model.v2

/**
 * The course roles of a user
 */
enum class CourseRole {
    /**
     * Admin of the course.
     */
    DOCENT,

    /**
     * Moderator of the course
     */
    TUTOR,

    /**
     * Participant in the course.
     */
    STUDENT;

    /**
     * Parse an int to a course role.
     * @param roleId The role id
     * @return Course role, where the id is mapped to its assigned enum type, if no enum type
     *         is assigned to the roleId provided, then the STUDENT type is returned.
     */
    fun parse(roleId: Int): CourseRole {
        return when (roleId) {
            0 -> DOCENT
            1 -> TUTOR
            else -> STUDENT
        }
    }
    /**
     * Parse a string to a course role.
     * @param roleName The role name
     * @return Course role, where the name is mapped to its assigned enum type, if no enum type
     *         is assigned to the name provided, then the STUDENT type is returned.
     */
    fun parse(roleName: String): CourseRole {
        return when (roleName) {
            "DOCENT" -> DOCENT
            "TUTOR" -> TUTOR
            else -> STUDENT
        }
    }
}
